src/logworker.cpp

```diff
@@ -120,6 +120,20 @@ namespace g3 {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, sink] {
+         for (auto it = _impl._sinks.begin(); it != _impl._sinks.end(); ) {
+            if ((*it) == sink) {
+               it = _impl._sinks.erase(it);
```

Comment:
Yes, I made a change which is similar to your suggestion. Please take a look ;) I also changed the way sink shared_ptr is passed to lambdas in add/removeSink methods. I switched to weak_ptr instead to prevent keeping shared_ptr instances as captured lambda values. This allows making simple and clear test cases for this functionality; otherwise sink cannot be destroyed easily ;)